src/main/java/screens/WeekViewScreen.java

```diff
@@ -0,0 +1,304 @@
+
+            screens.get("click task").setVisible(true);
+        }
+    }
+
+    /**
+     * Invoked the first time a window is made visible.
+     *
+     * @param e the event to be processed
+     */
+    @Override
+    public void windowOpened(WindowEvent e) {
+
+    }
+
+    /**
+     * Invoked when the user attempts to close the window
+     * from the window's system menu.
+     *
+     * @param e the event to be processed
+     */
+    @Override
+    public void windowClosing(WindowEvent e) {
+        int confirmed = JOptionPane.showConfirmDialog(null,
+                "Are you sure you want to exit the program?", "Exit Program Message Box",
+                JOptionPane.YES_NO_OPTION);
+
+        if (confirmed == JOptionPane.YES_OPTION) {
+            this.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        } else {
+            this.setDefaultCloseOperation(JFrame.DO_NOTHING_ON_CLOSE);
+        }
+    }
+
+    /**
+     * Invoked when a window has been closed as the result
+     * of calling dispose on the window.
+     *
+     * @param e the event to be processed
+     */
+    @Override
+    public void windowClosed(WindowEvent e) {
+
+    }
+
+    /**
+     * Invoked when a window is changed from a normal to a
+     * minimized state. For many platforms, a minimized window
+     * is displayed as the icon specified in the window's
+     * iconImage property.
+     *
+     * @param e the event to be processed
+     * @see Frame#setIconImage
+     */
+    @Override
+    public void windowIconified(WindowEvent e) {
+
+    }
+
+    /**
+     * Invoked when a window is changed from a minimized
+     * to a normal state.
+     *
+     * @param e the event to be processed
+     */
+    @Override
+    public void windowDeiconified(WindowEvent e) {
+
+    }
+
+    /**
+     * Invoked when the Window is set to be the active Window. Only a Frame or
+     * a Dialog can be the active Window. The native windowing system may
+     * denote the active Window or its children with special decorations, such
+     * as a highlighted title bar. The active Window is always either the
+     * focused Window, or the first Frame or Dialog that is an owner of the
+     * focused Window.
+     *
+     * @param e the event to be processed
+     */
+    @Override
+    public void windowActivated(WindowEvent e) {
+
+    }
+
+    /**
+     * Invoked when a Window is no longer the active Window. Only a Frame or a
+     * Dialog can be the active Window. The native windowing system may denote
+     * the active Window or its children with special decorations, such as a
+     * highlighted title bar. The active Window is always either the focused
+     * Window, or the first Frame or Dialog that is an owner of the focused
+     * Window.
+     *
+     * @param e the event to be processed
+     */
+    @Override
+    public void windowDeactivated(WindowEvent e) {
```

Comment:
Kindly check if you are required to implement these interface methods, since all of them have no functionality.